spec/commands/open.spec.ts

```diff
@@ -0,0 +1,206 @@
+        parse: args,
+        argv: ["open", "schema", "modelA", `--app=${testApp.slug}`, `--env=${testApp.environments[0]!.name}`],
+      });
+
+      await openCommand(ctx);
+
+      expect(open).toHaveBeenCalledWith(`https://${testApp.primaryDomain}/edit/${testApp.environments[0]!.name}/model/modelA/schema`);
+
+      expectStdout().toMatchInlineSnapshot(`
+          "Opened schema viewer for environment development for model modelA please check your browser.
+          "
+        `);
+    });
+
+    it("to the model's data page based on user selection from a list of available models if --show-all flag is passed", async () => {
+      mock(user, "getUser", () => testUser);
+      nockGetModels();
+
+      mock(select, () => "user");
+
+      ctx = makeContext({
+        parse: args,
+        argv: ["open", "data", "--show-all", `--app=${testApp.slug}`, `--env=${testApp.environments[0]!.name}`],
+      });
+
+      await openCommand(ctx);
+
+      expectStdout().toMatchInlineSnapshot(`
+          "Opened data viewer for environment development for model user please check your browser.
+          "
+        `);
+    });
+
+    it("to the model's schema page based on user selection from a list of available models if --show-all flag is passed", async () => {
+      mock(user, "getUser", () => testUser);
+      nockGetModels();
+
+      mock(select, () => "user");
+
+      ctx = makeContext({
+        parse: args,
+        argv: ["open", "schema", "--show-all", `--app=${testApp.slug}`, `--env=${testApp.environments[0]!.name}`],
+      });
+
+      await openCommand(ctx);
+
+      expectStdout().toMatchInlineSnapshot(`
+        "Opened schema viewer for environment development for model user please check your browser.
+        "
+      `);
+    });
+  });
+
+  it("displays the possible subcommands when one is not passed", async () => {
+    const error = await expectError(() => openCommand(ctx));
+
+    expect(error).toBeInstanceOf(ArgError);
+    expect(error.message).toMatchInlineSnapshot(`
+      "
+      Missing subcommand for ggt open [subcommand].
+      Run ggt open -help for more information or pass in a subcommand:
+
+        • logs
+        • data
+        • schema
+        • permissions"
+    `);
+  });
+
+  it("displays the closest match for a model that does not exist", async () => {
+    mock(user, "getUser", () => testUser);
+    nockGetModels();
+
+    ctx = makeContext({
+      parse: args,
+      argv: ["open", "data", "use", `--app=${testApp.slug}`, `--env=${testApp.environments[0]!.name}`],
+    });
+
+    await openCommand(ctx);
+
+    expectStdout().toMatchInlineSnapshot(`
+      "      Unknown model use
+
+            Did you mean user?
+            
+            Run ggt open --help for usage or run command with --show-all to see all available models
+      "
+    `);
+  });
+});
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+const nockGetModels = () => {
+  return nockEditResponse({
```

Comment:
nit: I think `nockEditReponse` is kinda weird here; maybe we should change this to `nockApiReponse`

Reply:
ya i think that name would make more sense in this case, maybe we can rename this to just `nockGraphqlResponse` to be more general and make sense for nocking both edit and api endpoints? @scott-rc

Reply:
If we go with [the `Api` idea](https://github.com/gadget-inc/ggt/pull/1241/files#r1492836757), how about we:
1. Rename `nockEditResponse` to `nockGraphQLResponse` and make it take an endpoint
2. Re-create `nockEditResponse` that simply wraps `nockGraphQLResponse` and passed the correct endpoint
3. Create `nockApiResponse` that simply wraps `nockGraphQLResponse` and passed the correct endpoint

Make sense?